### PR TITLE
Refine deployment docs, add advanced strategies

### PR DESCRIPTION
**Pull request by ahardin-rh**
continuation of https://github.com/openshift/openshift-docs/pull/2890
applied updated style guidelines per https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc

### CODE REVIEW

**Comment by kargakis** [comment]
@mfojtik ptal

**Comment by ahardin-rh** [comment]
Preview Build: http://file.rdu.redhat.com/~ahardin/11022016/2890-deployment/dev_guide/deployments.html

**Comment by mfojtik** [comment]
LGTM

File: dev_guide/deployments.adoc [hunk 1]
```diff
@@ -829,3 +952,283 @@ spec:
...
   serviceAccountName: <service_account>
 ----
+
+[[advanced-deployment-strategies]]
+== Advanced Deployment Strategies
+
+[[advanced-deployment-strategies-blue-green-deployments]]
+=== Blue-Green Deployment
+
+link:http://martinfowler.com/bliki/BlueGreenDeployment.html[Blue-Green
```
Comment (adellape) [inline dev_guide/deployments.adoc]: Let's make this link ifdef'd only for Origin for now?

File: dev_guide/deployments.adoc [hunk 2]
```diff
@@ -829,3 +952,283 @@ spec:
...
+deployments] involve running two versions of an application at the same time and
+moving production traffic from the old version to the new version. There are
+several ways to implement a blue-green deployment in {product-title}.
+
+[[advanced-deployment-strategies-when-to-use-blue-green-deployment]]
+=== When to Use a Blue-Green Deployment
+
+Use a Blue-Green deployment When you want to test a new version of your
```
Comment (adellape) [inline dev_guide/deployments.adoc]: s/When/when/

File: dev_guide/deployments.adoc [hunk 3]
```diff
@@ -829,3 +952,283 @@ spec:
...
+application in a production environment before moving traffic to it.
+
+Blue-Green deployments make switching between two different versions of your
```
Comment (adellape) [inline dev_guide/deployments.adoc]: Sometimes it's init capped "Blue-Green" and elsewhere it's lowercase "blue-green". Pick one and sync up on it? I'd lean towards lowercase.

File: dev_guide/deployments.adoc [hunk 4]
```diff
@@ -829,3 +952,283 @@ spec:
...
+
+An A/B deployment groups different configuration and code -- multiple shards --
+together under a single logical endpoint. Generally, these deployments, if they
+access persistent data, should properly deal with N-1 compatibility (the more
+shards you have, the more possible versions you have running). Use this pattern
+when you need separate internal configuration and code, but end users should not
+be aware of the changes.
+
+[[advanced-deployment-a-b-deployment-example]]
+=== A/B deployment Example
```
Comment (adellape) [inline dev_guide/deployments.adoc]: s/deployment/Deployment/

File: dev_guide/deployments.adoc [hunk 5]
```diff
@@ -829,3 +952,283 @@ spec:
...
+environment variable in either deployment config `oc edit dc/ab-example-a` or
+`oc edit dc/ab-example-b`. You can add additional shards by repeating steps 5-7.
++
+[NOTE]
+====
+These steps will be simplified in future versions of {product-title}.
+====
+
+[[shard-proxy-traffic-splitter]]
+=== Shard Proxy / Traffic Splitter
```
Comment (adellape) [inline dev_guide/deployments.adoc]: We call it a "Shard Proxy" and a "proxy shard"? Same thing I assume, different way of referring to it?

**Review by adellape — COMMENTED** [review]
Some minor things inline.

**Comment by adellape** [comment]
@kargakis See https://bugzilla.redhat.com/show_bug.cgi?id=1384018#c2 for my plans to continue iterating after this 3149 PR is merged. But don't let it stop you from anything you were going to submit per your comments in:

https://trello.com/c/AZ1F8LuM/700-3-integrate-upstream-deployments#comment-57fe1d156ad73f8d694ddb9e
https://trello.com/c/oAlPryPh/793-3-need-to-document-how-to-use-upstream-deployments#comment-57e3ffa1ab94ac1427d205e9

Just ping me if you open a new PR. :bowing_man:

File: dev_guide/deployments.adoc [hunk 6]
```diff
@@ -829,3 +952,289 @@ spec:
...
+affected. You can easily trigger a deployment by changing the `SUBTITLE`
+environment variable in either deployment config `oc edit dc/ab-example-a` or
+`oc edit dc/ab-example-b`. You can add additional shards by repeating steps 5-7.
++
+[NOTE]
+====
+These steps will be simplified in future versions of {product-title}.
+====
+
+[[shard-proxy-traffic-splitter]]
```
Comment (adellape) [inline dev_guide/deployments.adoc]: Sync up new title here? `[[proxy-shard-traffic-splitter]]`

**Review by adellape — APPROVED** [review]
One tiny nit, but LGTM!

**Comment by ahardin-rh** [comment]
@adellape Fixed! Merging now.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../dev_guide/deployments.adoc#dev-guide-deployments[Deployments]
|Added a new xref:../dev_guide/deployments.adoc#advanced-deployment-strategies[Advanced Deployment Strategies] section.
%

**Pull request merged**